apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V53__preadj_mbi_caching-1.sql

```diff
@@ -0,0 +1,37 @@
+/*
+ * Table to hold known MBI and their hashed equivalents.
+ */
+CREATE TABLE "pre_adj"."MbiCache" (
+    "mbiId"   bigint      NOT NULL PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    "mbi"     VARCHAR(13) NOT NULL,
+    "hash"    VARCHAR(64) NOT NULL,
+    "oldHash" VARCHAR(64),
+    "lastUpdated" timestamp with time zone
+);
+
+/*
+ * We need these indexes to support queries by mbi, hash, and oldHash.  Note that oldHash
+ * is usually null (only set to a value during hash rotation) so the index cannot be unique.
+ */
+CREATE UNIQUE INDEX "MbiCache_mbi_idx" on "pre_adj"."MbiCache"("mbi");
+CREATE UNIQUE INDEX "MbiCache_hash_idx" on "pre_adj"."MbiCache"("hash");
```

Comment:
I believe postgres does allow you to have multiple nulls in a unique column, however I'm wondering if we'd ever run into collisions when copying a value from old to new or vice versa.  Would it be best to just make the mbi column unique?

Reply:
These are both NOT NULL columns.  A collision in the hash column would be a serious problem that I think we'd want to detect early.  The UNIQUE constraint would identify it for us quickly.

Reply:
The first part of my comment was in reference to line 18, the oldHash column

Reply:
I don't thik UNIQUE is allowed for NULLABLE columns though.

Reply:
Wow.  I'm wrong about UNIQUE on NULLABLE.  Just tried it and it worked.  That is really odd...

Reply:
With that said, maybe just remove the comment on line 14 since thats not the case